R/tmb-methods.R

```diff
@@ -427,3 +427,50 @@ h_residuals_response <- function(object) {
   assert_class(object, "mmrm_tmb")
   component(object, "y_vector") - unname(fitted(object))
 }
+
+
+#' @describeIn mmrm_tmb_methods Simulates from an mmrm model object
+#' @importFrom stats simulate
+#' @exportS3Method
+#'
+#' @param object an object of class 'mmrm'
+#' @param nsim number of response vectors to simulate. Defaults to 1.
+#' @param seed an object specifying if and how the random number generator should be initialized (‘seeded’).
+#' @param newdata data frame for which to evaluate predictions.
+#' @param ...
+#'
+#' @return A \code{data.frame} of dimension [n, m] where n is the number of observations
+#' fitted in the model, and m is the number \code{nsim} of simulated responses.
+simulate.mmrm_tmb <- function(object, nsim = 1,
+                          seed = NULL, newdata = object$data,
+                          ...){
+  if(!is.null(seed)) set.seed(seed)
+
+  # build data.frame of prediction vectors
+  ret <- predict(object, newdata = newdata, se.fit = FALSE, interval = "none")
```

Comment:
I think `newdata` here is confusing that it is only used for the prediction; the covariance is not taken into consideration. It can be true that user provided new data does not match the old data (not ordered by subject and visit etc) can cause issues in the next steps

Reply:
e.g. newdata and fit$tmb_data do not have the same dimension

Reply:
The point about data not ordered by subject/visit is a good one -- I'll address this in the next commit and add a test for differently ordered data.